mllib/src/main/scala/org/apache/spark/ml/param/params.scala

```diff
@@ -254,21 +251,69 @@ object ParamValidators {
   }
 
   /**
-   * Checks that only one of the params passed as arguments is set. If this is not true, an
-   * `IllegalArgumentException` is raised.
+   * Utility for Param validity checks for Transformers which have both single- and multi-column
+   * support.  This utility assumes that `inputCol` indicates single-column usage and
+   * that `inputCols` indicates multi-column usage.
+   *
+   * This checks to ensure that exactly one set of Params has been set, and it
+   * raises an `IllegalArgumentException` if not.
+   *
+   * @param singleColumnParams Params which should be set (or have defaults) if `inputCol` has been
+   *                           set.  This does not need to include `inputCol`.
+   * @param multiColumnParams Params which should be set (or have defaults) if `inputCols` has been
+   *                           set.  This does not need to include `inputCols`.
    */
-  def checkExclusiveParams(model: Params, params: String*): Unit = {
-    val (existingParams, nonExistingParams) = params.partition(model.hasParam)
-    if (nonExistingParams.nonEmpty) {
-      val pronoun = if (nonExistingParams.size == 1) "It" else "They"
-      LOGGER.warn(s"Ignored ${nonExistingParams.mkString("`", "`, `", "`")} while checking " +
-        s"exclusive params. $pronoun don't exist for the specified model the model.")
+  def checkSingleVsMultiColumnParams(
+      model: Params,
+      singleColumnParams: Seq[Param[_]],
+      multiColumnParams: Seq[Param[_]]): Unit = {
+    val name = s"${model.getClass.getSimpleName} $model"
+
+    def checkExclusiveParams(
+        isSingleCol: Boolean,
+        requiredParams: Seq[Param[_]],
+        excludedParams: Seq[Param[_]]): Unit = {
+      val badParamsMsgBuilder = new mutable.StringBuilder()
```

Comment:
This builder lets us include all incorrectly set Params in the error message, rather than just one.